src/Adapter/MSTest.TestAdapter/RunConfigurationSettings.cs

```diff
@@ -155,4 +159,39 @@ private static RunConfigurationSettings ToSettings(XmlReader reader)
 
         return settings;
     }
+
+    private static RunConfigurationSettings ToSettings(IConfiguration configuration)
+    {
+        // Expected format of the json is: -
+        // "mstest" : {
+        //  "runConfiguration": {
+        //    "collectSourceInformation": true,
+        //    "executionApartmentState": "STA"
+        //  }
+        // }
+
+        // Initialize settings object
+        RunConfigurationSettings settings = new();
+
+        if (bool.TryParse(configuration["mstest:runConfiguration:collectSourceInformation"], out bool collectSourceInformation))
+        {
+            settings.CollectSourceInformation = collectSourceInformation;
+            PlatformServiceProvider.Instance.AdapterTraceLogger.LogInfo(
+                "CollectSourceInformation value Found : {0}", collectSourceInformation);
+        }
+
+        string? apartmentStateValue = configuration["mstest:runConfiguration:executionApartmentState"];
+        if (!string.IsNullOrEmpty(apartmentStateValue) &&
```

Comment:
I think that the `Enum.TryParse` is enough, I suppose that if you pass a null to it will return `false`.

Reply:
when i remove it some tests fails